Structured Logging migration: modify volume and container part logs o…

yangjunmyfm192085 opened the pull request:
…f kubelet.

Signed-off-by: JunYang <yang.jun22@zte.com.cn>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind feature
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:
Structured Logging migration: modify volume and container part logs of kubelet.
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md
**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:
part of kubernetes/enhancements#1602 
As it is related to log, change the log here to structured log.
**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
yes
For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

yangjunmyfm192085 committed:
Structured Logging migration: modify volume and container part logs of kubelet.

Signed-off-by: JunYang <yang.jun22@zte.com.cn>

matthyx commented:
/cc @serathius 
/cc @matthyx

serathius commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -117,7 +117,7 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 		// kl.getPodVolumePathListFromDisk(). Can this be cleaned up?
 		if podVolumesExist := kl.podVolumesExist(uid); podVolumesExist {
-			klog.V(3).Infof("Orphaned pod %q found, but volumes are not cleaned up", uid)
+			klog.V(3).InfoS("The orphaned pod found, but volumes are not cleaned up", "uid", uid)
```
```suggestion
			klog.V(3).InfoS("The orphaned pod found, but volumes are not cleaned up", "podUID", uid)
```

serathius commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -142,18 +142,18 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 		}
 
-		klog.V(3).Infof("Orphaned pod %q found, removing", uid)
+		klog.V(3).InfoS("The orphaned pod found, removing", "uid", uid)
```
```suggestion
		klog.V(3).InfoS("The orphaned pod found, removing", "podUID", uid)
```

serathius commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -142,18 +142,18 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
+		klog.V(3).InfoS("The orphaned pod found, removing", "uid", uid)
 		if err := removeall.RemoveAllOneFilesystem(kl.mounter, kl.getPodDir(uid)); err != nil {
-			klog.Errorf("Failed to remove orphaned pod %q dir; err: %v", uid, err)
+			klog.ErrorS(err, "Failed to remove the orphaned pod dir", "uid", uid)
```
```suggestion
			klog.ErrorS(err, "Failed to remove the orphaned pod dir", "podUID", uid)
```

serathius commented on pkg/kubelet/pod_container_deletor.go:
```diff
@@ -49,7 +49,7 @@ func newPodContainerDeletor(runtime kubecontainer.Runtime, containersToKeep int)
 			id := <-buffer
 			if err := runtime.DeleteContainer(id); err != nil {
-				klog.Warningf("[pod_container_deletor] DeleteContainer returned error for (id=%v): %v", id, err)
+				klog.InfoS("[pod_container_deletor] DeleteContainer returned error", "containerId", id, "error", err)
```
```suggestion
				klog.InfoS("DeleteContainer returned error", "containerID", id, "err", err)
```

serathius commented on pkg/kubelet/pod_container_deletor.go:
```diff
@@ -110,7 +110,7 @@ func (p *podContainerDeletor) deleteContainersInPod(filterContainerID string, po
 		case p.worker <- candidate.ID:
 		default:
-			klog.Warningf("Failed to issue the request to remove container %v", candidate.ID)
+			klog.InfoS("Failed to issue the request to remove container", "containerId", candidate.ID)
```
```suggestion
			klog.InfoS("Failed to issue the request to remove container", "containerID", candidate.ID)
```

serathius commented on pkg/kubelet/pod_workers.go:
```diff
@@ -188,7 +188,7 @@ func (p *podWorkers) managePodLoop(podUpdates <-chan UpdatePodOptions) {
 		if err != nil {
 			// IMPORTANT: we do not log errors here, the syncPodFn is responsible for logging errors
-			klog.Errorf("Error syncing pod %s (%q), skipping: %v", update.Pod.UID, format.Pod(update.Pod), err)
+			klog.ErrorS(err, "Error syncing pod, skipping", "pod", format.Pod(update.Pod), "podUID", update.Pod.UID)
```
```suggestion
			klog.ErrorS(err, "Error syncing pod, skipping", "pod", klog.KObj(update.Pod), "podUID", update.Pod.UID)
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -51,15 +51,15 @@ func (kl *Kubelet) RunOnce(updates <-chan kubetypes.PodUpdate) ([]RunPodResult,
 	if _, err := os.Stat(ContainerLogsDir); err != nil {
 		if err := kl.os.MkdirAll(ContainerLogsDir, 0755); err != nil {
-			klog.Errorf("Failed to create directory %q: %v", ContainerLogsDir, err)
+			klog.ErrorS(err, "Failed to create directory", "directory", ContainerLogsDir)
```
```suggestion
			klog.ErrorS(err, "Failed to create directory", "path", ContainerLogsDir)
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -51,15 +51,15 @@ func (kl *Kubelet) RunOnce(updates <-chan kubetypes.PodUpdate) ([]RunPodResult,
 	select {
 	case u := <-updates:
-		klog.Infof("processing manifest with %d pods", len(u.Pods))
+		klog.InfoS("processing manifest with pods", "podsNum", len(u.Pods))
```
```suggestion
		klog.InfoS("Processing manifest with pods", "podsNum", len(u.Pods))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -51,15 +51,15 @@ func (kl *Kubelet) RunOnce(updates <-chan kubetypes.PodUpdate) ([]RunPodResult,
+		klog.InfoS("processing manifest with pods", "podsNum", len(u.Pods))
 		result, err := kl.runOnce(u.Pods, runOnceRetryDelay)
-		klog.Infof("finished processing %d pods", len(u.Pods))
+		klog.InfoS("finished processing pods", "podsNum", len(u.Pods))
```
```suggestion
		klog.InfoS("Finished processing pods", "podsNum", len(u.Pods))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
 			faliedContainerName, err := kl.getFailedContainers(res.Pod)
 			if err != nil {
-				klog.Infof("unable to get failed containers' names for pod %q, error:%v", format.Pod(res.Pod), err)
+				klog.InfoS("unable to get failed containers' names for pod", "pod", format.Pod(res.Pod), "error", err)
```
```suggestion
				klog.InfoS("Unable to get failed containers' names for pod", "pod", format.Pod(res.Pod), "err", err)
```

serathius commented on pkg/kubelet/container/container_gc.go:
```diff
@@ -82,6 +82,6 @@ func (cgc *realContainerGC) GarbageCollect() error {
 
 func (cgc *realContainerGC) DeleteAllUnusedContainers() error {
-	klog.Infof("attempting to delete unused containers")
+	klog.InfoS("attempting to delete unused containers")
```
```suggestion
	klog.InfoS("Attempting to delete unused containers")
```

serathius commented on pkg/kubelet/container/helpers.go:
```diff
@@ -82,13 +82,13 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 	// Check RestartPolicy for dead container
 	if pod.Spec.RestartPolicy == v1.RestartPolicyNever {
-		klog.V(4).Infof("Already ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
+		klog.V(4).InfoS("Already ran container, do nothing", "pod", format.Pod(pod), "containerName", container.Name)
```
```suggestion
		klog.V(4).InfoS("Already ran container, do nothing", "pod", klog.KObjod(pod), "containerName", container.Name)
```

serathius commented on pkg/kubelet/container/helpers.go:
```diff
@@ -82,13 +82,13 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 		// Check the exit code.
 		if status.ExitCode == 0 {
-			klog.V(4).Infof("Already successfully ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
+			klog.V(4).InfoS("Already successfully ran container, do nothing", "pod", format.Pod(pod), "containerName", container.Name)
```
```suggestion
			klog.V(4).InfoS("Already successfully ran container, do nothing", "pod", klog.KObj(pod), "containerName", container.Name)
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
+				klog.InfoS("unable to get failed containers' names for pod", "pod", format.Pod(res.Pod), "error", err)
 			} else {
-				klog.Infof("unable to start pod %q because container:%v failed", format.Pod(res.Pod), faliedContainerName)
+				klog.InfoS("unable to start pod because container failed", "pod", format.Pod(res.Pod), "containerName", faliedContainerName)
```
```suggestion
				klog.InfoS("Unable to start pod because container failed", "pod", klog.KObj(res.Pod), "containerName", faliedContainerName)
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
 			failedPods = append(failedPods, format.Pod(res.Pod))
 		} else {
-			klog.Infof("started pod %q", format.Pod(res.Pod))
+			klog.InfoS("started pod", "pod", format.Pod(res.Pod))
```
```suggestion
			klog.InfoS("started pod", "pod", klog.KObj(res.Pod))
```

serathius replied:
```suggestion
			klog.InfoS("Started pod", "pod", klog.KObj(res.Pod))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
 		return results, fmt.Errorf("error running pods: %v", failedPods)
 	}
-	klog.Infof("%d pods started", len(pods))
+	klog.InfoS("pods started", "podNum", len(pods))
```
```suggestion
	klog.InfoS("Pods started", "podNum", len(pods))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -120,14 +120,14 @@ func (kl *Kubelet) runPod(pod *v1.Pod, retryDelay time.Duration) error {
 
 		if kl.isPodRunning(pod, status) {
-			klog.Infof("pod %q containers running", format.Pod(pod))
+			klog.InfoS("pod's containers running", "pod", format.Pod(pod))
```
```suggestion
			klog.InfoS("Pod's containers running", "pod", klog.KObj(pod))
```

serathius replied:
```suggestion
			klog.InfoS("Pod's containers running", "pod", klog.KObj(pod))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
-				klog.Infof("unable to start pod %q because container:%v failed", format.Pod(res.Pod), faliedContainerName)
+				klog.InfoS("unable to start pod because container failed", "pod", format.Pod(res.Pod), "containerName", faliedContainerName)
 			}
 			failedPods = append(failedPods, format.Pod(res.Pod))
```
```suggestion
			failedPods = append(failedPods, klog.KObj(res.Pod))
```

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -120,14 +120,14 @@ func (kl *Kubelet) runPod(pod *v1.Pod, retryDelay time.Duration) error {
 			return nil
 		}
-		klog.Infof("pod %q containers not running: syncing", format.Pod(pod))
+		klog.InfoS("pod's containers not running: syncing", "pod", format.Pod(pod))
```
```suggestion
		klog.InfoS("Pod's containers not running: syncing", "pod", klog.KObj(pod))
```

serathius commented:
Ok, please apply fixes to rest of lines based on rules:
* Log message should start from capital letter
* Replace `format.Pod` with `klog.KObj`
* Use `podUID` over `uid` for pods 

Feel free to ping me when your done.

matthyx commented:
/ok-to-test

yangjunmyfm192085 commented:
> * g message should start from capital letter
> * Replace `format.Pod` with `klog.KObj`
> * Use `podUID` over `uid` for pods
> 
> Feel free to ping me when your done.

Thanks,I'll check again.

yangjunmyfm192085 commented:
/retest

adisky commented on pkg/kubelet/container/helpers.go:
```diff
@@ -82,13 +82,13 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 	// Check RestartPolicy for dead container
 	if pod.Spec.RestartPolicy == v1.RestartPolicyNever {
-		klog.V(4).Infof("Already ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
+		klog.V(4).InfoS("Already ran container, do nothing", "pod", klog.KObjod(pod), "containerName", container.Name)
```
klog.KObj(pod)

yangjunmyfm192085 replied:
Thanks, I have fixed it.

yangjunmyfm192085 commented:
/test pull-kubernetes-verify

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-gce-storage-snapshot
/test pull-kubernetes-e2e-gce-csi-serial

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -525,11 +502,7 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
-			pod.Namespace,
-			pvcSource.ClaimName)
-
+		klog.V(5).InfoS("Found PVC", "podNamespace", pod.Namespace, "ClaimName", pvcSource.ClaimName)
```
```suggestion
		klog.V(5).InfoS("Found PVC", "pvc",klog.KRef(pod.Namespace, pvcSource.ClaimName))
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -425,8 +403,7 @@ func (dswp *desiredStateOfWorldPopulator) checkVolumeFSResize(
 		// This volume is used as read only by this pod, we don't perform resize for read only volumes.
-		klog.V(5).Infof("Skip file system resize check for volume %s in pod %s/%s "+
-			"as the volume is mounted as readonly", podVolume.Name, pod.Namespace, pod.Name)
+		klog.V(5).InfoS("Skip file system resize check for the volume,as the volume is mounted as readonly", "pod", klog.KObj(pod), "volumeName", podVolume.Name)
```
```suggestion
		klog.V(5).InfoS("Skip file system resize check for the volume, as the volume is mounted as readonly", "pod", klog.KObj(pod), "volumeName", podVolume.Name)
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -349,20 +336,11 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
-				podVolume.Name,
-				volumeSpec.Name(),
-				uniquePodName)
+			klog.V(4).InfoS("Added volume to desired state.", "podName", uniquePodName, "volumeName", podVolume.Name, "volumeSpecName", volumeSpec.Name())
```
```suggestion
			klog.V(4).InfoS("Added volume to desired state", "podName", uniquePodName, "volumeName", podVolume.Name, "volumeSpecName", volumeSpec.Name())
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -278,19 +272,16 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
-			klog.V(4).Infof(
-				volumeToMount.GenerateMsgDetailed(fmt.Sprintf("Actual state has not yet has this volume mounted information and pod (%q) still exists in pod manager, skip removing volume from desired state",
-					format.Pod(volumeToMount.Pod)), ""))
+			klog.V(4).InfoS(volumeToMount.GenerateMsgDetailed(fmt.Sprintf("Actual state has not yet has this volume mounted information and pod (%q) still exists in pod manager, skip removing volume from desired state",
```
Please extract all information hat volumeToMount.GenerateMsgDetailed and put it as normal keysto InfoS method.

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -278,19 +272,16 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 			continue
 		}
-		klog.V(4).Infof(volumeToMount.GenerateMsgDetailed("Removing volume from desired state", ""))
+		klog.V(4).InfoS(volumeToMount.GenerateMsgDetailed("Removing volume from desired state", ""))
```
Ditto

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -171,11 +171,7 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoop() {
-			dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration),
-			dswp.getPodStatusRetryDuration)
-
+		klog.V(5).InfoS("Skipping findAndRemoveDeletedPods(). ", "nextRetryTime", dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration), "RetryDuration", dswp.getPodStatusRetryDuration)
```
```suggestion
		klog.V(5).InfoS("Skipping findAndRemoveDeletedPods(). ", "nextRetryTime", dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration), "retryDuration", dswp.getPodStatusRetryDuration)
```

serathius commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -548,12 +544,7 @@ func (asw *actualStateOfWorld) MarkRemountRequired(
-					podObj.podUID,
-					volumeObj.volumeName,
-					podObj.volumeSpec.Name())
+				klog.ErrorS(nil, "MarkRemountRequired failed to FindPluginBySpec for volume", "podName", podObj.podName, "podUID", podObj.podUID, "volumeName", volumeName, "volumeSpecName", podObj.volumeSpec.Name())
```
```suggestion
				klog.ErrorS(nil, "MarkRemountRequired failed to FindPluginBySpec for volume", "pod", klog.KObj(podObj), "podUID", podObj.podUID, "volumeName", volumeName, "volumeSpecName", podObj.volumeSpec.Name())
```

serathius replied:
Prefer using pod object over extracting a name from it.

yangjunmyfm192085 commented:
/test pull-kubernetes-verify

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -142,7 +142,7 @@ func (kl *Kubelet) runPod(pod *v1.Pod, retryDelay time.Duration) error {
 		}
 		// TODO(proppy): health checking would be better than waiting + checking the state at the next iteration.
-		klog.Infof("pod %q containers synced, waiting for %v", format.Pod(pod), delay)
+		klog.InfoS("Pod's containers synced, waiting", "pod", klog.KObj(pod), "waitingTime", delay)
```
```suggestion
		klog.InfoS("Pod's containers synced, waiting", "pod", klog.KObj(pod), "delay", delay)
```

serathius commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -548,12 +544,7 @@ func (asw *actualStateOfWorld) MarkRemountRequired(
-					podObj.podUID,
-					volumeObj.volumeName,
-					podObj.volumeSpec.Name())
+				klog.ErrorS(nil, "MarkRemountRequired failed to FindPluginBySpec for volume", "pod", klog.KObj(podObj), "podUID", podObj.podUID, "volumeName", volumeName, "volumeSpecName", podObj.volumeSpec.Name())
```
Looks like podObj is not an standard API object that implements interface required by `klog.KObj`, Can you check if it has field `podNamespace` if so let's use `klog.KRef`, if not we should go back to `"podName", podObj.podName`.

yangjunmyfm192085 replied:
Ok,I will check it.

serathius commented:
Could you [squash commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits)?

yangjunmyfm192085 commented:
> Could you [squash commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits)?

yeah

yangjunmyfm192085 commented:
> Could you [squash commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits)?

yeah, Is it not ok?

serathius commented:
PR still has 26 commits, most of them are updates from review. From documentation linked before
```
All commits left on your branch after a review should represent meaningful milestones or units of work. Use commits to add clarity to the development and review process.
```
Could you squash commits so we have 1 commit left? Let me know if you need help with it. Happy to lend a hand.

yangjunmyfm192085 commented:
> commits

yeah,I modified some commits locally.
But I think I have fixed all of them.
Now, I really I have a question,
test pull-kubernetes-verify keep failing,I don't why?

yangjunmyfm192085 commented:
> PR still has 26 commits, most of them are updates from review. From documentation linked before
> 
> ```
> All commits left on your branch after a review should represent meaningful milestones or units of work. Use commits to add clarity to the development and review process.
> ```
> 
> Could you squash commits so we have 1 commit left? Let me know if you need help with it. Always happy to lend a hand.
I will check them once again.Make sure all commits have been processed

yangjunmyfm192085 commented:
/pull-kubernetes-e2e-kind-ipv6

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-kind

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-gce-csi-serial

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-gce-storage-snapshot

ehashman commented:
/triage accepted
/priority important-soon
/remove-kind feature
/kind cleanup

ehashman commented:
@yangjunmyfm192085 Logging changes require a release note. Please add a release note in your first comment corresponding to the files where log entries have been migrated.

/hold

yangjunmyfm192085 commented:
/retest

yangjunmyfm192085 commented:
/retest

adisky commented:
@yangjunmyfm192085 Please follow this doc to squash commits.
https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits

yangjunmyfm192085 commented:
> @yangjunmyfm192085 Please follow this doc to squash commits.
> https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits

Thank you very much. I think i 

> @yangjunmyfm192085 Please follow this doc to squash commits.
> https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits



> @yangjunmyfm192085 Please follow this doc to squash commits.
> https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits

Thank you very much. I do use this method to squash commits.

adisky commented:
> > @yangjunmyfm192085 Please follow this doc to squash commits.
> > https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits
> 
> Thank you very much. I do use this method to squash commits.

If you have squashed and pushed here you will see only one commit https://github.com/kubernetes/kubernetes/pull/98850/commits

yangjunmyfm192085 commented:
> > > @yangjunmyfm192085 Please follow this doc to squash commits.
> > > https://www.kubernetes.dev/docs/guide/github-workflow/#squash-commits
> > 
> > 
> > Thank you very much. I do use this method to squash commits.
> 
> If you have squashed and pushed here you will see only one commit https://github.com/kubernetes/kubernetes/pull/98850/commits


Thanks,I'll deal with it right away.

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -278,19 +271,19 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 			continue
 		}
-		klog.V(4).Infof(volumeToMount.GenerateMsgDetailed("Removing volume from desired state", ""))
+		klog.V(4).InfoS("Removing volume from desired state", "pod", klog.KObj(volumeToMount.Pod), "PodUID", volumeToMount.Pod.UID, "volumeName", volumeToMountSpecName)
```
```suggestion
		klog.V(4).InfoS("Removing volume from desired state", "pod", klog.KObj(volumeToMount.Pod), "podUID", volumeToMount.Pod.UID, "volumeName", volumeToMountSpecName)
```

yangjunmyfm192085 commented:
/retest

yangjunmyfm192085 commented:
/retest

navidshaikh commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -652,23 +649,23 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 				continue
 			}
-			klog.V(4).Infof("Volume: %s (pod UID %s) is marked device as mounted and added into the actual state", volume.volumeName, volume.podName)
+			klog.V(4).InfoS("The olume is marked device as mounted and added into the actual state", "podName", volume.podName, "volumeName", volume.volumeName)
```
```suggestion
			klog.V(4).InfoS("The volume is marked device as mounted and added into the actual state", "podName", volume.podName, "volumeName", volume.volumeName)
```

yangjunmyfm192085 commented:
/retest

matthyx commented:
@ehashman I think your hold has been adressed?
/lgtm
/assign @sjenning

ehashman commented:
Can we update the release note? It should just be a single line, and we don't need a doc update. As is this is going to create needless noise.

yangjunmyfm192085 commented:
> Can we update the release note? It should just be a single line, and we don't need a doc update. As is this is going to create needless noise.

Happy New Year.
I have fixed it ,Thank you very much.

serathius commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -62,11 +62,11 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 	}
 	if len(volumePaths) > 0 {
-		klog.V(4).Infof("pod %q found, but volumes are still mounted on disk %v", podUID, volumePaths)
+		klog.V(4).InfoS("The pod found, but volumes are still mounted on disk", "podUID", podUID, "volumePaths", volumePaths)
```
```suggestion
		klog.V(4).InfoS("The pod found, but volumes are still mounted on disk", "podUID", podUID, "paths", volumePaths)
```

yangjunmyfm192085 commented:
I have fixed all and the pr is ready for review @ehashman.

ehashman commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -404,11 +404,11 @@ func (rc *reconciler) syncStates() {
 				// Some pod needs the volume, don't clean it up and hope that
 				// reconcile() calls SetUp and reconstructs the volume in ASW.
-				klog.V(4).Infof("Volume exists in desired state (volume.SpecName %s, pod.UID %s), skip cleaning up mounts", volume.volumeSpecName, volume.podName)
+				klog.V(4).InfoS("Volume exists in desired state, skip cleaning up mounts", "podName", volume.podName, "volumeSpecName", volume.volumeSpecName)
```
```suggestion
				klog.V(4).InfoS("Volume exists in desired state, skip cleaning up mounts", "pod", klog.KObj(reconstructedVolume.pod), "volumeSpecName", volume.volumeSpecName)
```

ehashman commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -404,11 +404,11 @@ func (rc *reconciler) syncStates() {
 			}
 			// No pod needs the volume.
-			klog.Warningf("Could not construct volume information, cleanup the mounts. (pod.UID %s, volume.SpecName %s): %v", volume.podName, volume.volumeSpecName, err)
+			klog.InfoS("Could not construct volume information, cleanup the mounts.", "podName", volume.podName, "volumeSpecName", volume.volumeSpecName, "error", err)
```
```suggestion
			klog.InfoS("Could not construct volume information, cleaning up mounts", "pod", klog.KObj(reconstructedVolume.pod), "volumeSpecName", volume.volumeSpecName, "error", err)
```

ehashman commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -600,13 +597,13 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 				volume.devicePath = attachedVolume.DevicePath
 				volumesNeedUpdate[attachedVolume.Name] = volume
-				klog.V(4).Infof("Update devicePath from node status for volume (%q): %q", attachedVolume.Name, volume.devicePath)
+				klog.V(4).InfoS("Update devicePath from node status for volume", "volumeName", attachedVolume.Name, "devicePath", volume.devicePath)
```
```suggestion
				klog.V(4).InfoS("Update devicePath from node status for volume", "volumeName", attachedVolume.Name, "path", volume.devicePath)
```

@serathius should this just be path?

ehashman commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -728,6 +725,6 @@ func getVolumesFromPodDir(podDir string) ([]podVolume, error) {
 		}
 	}
-	klog.V(4).Infof("Get volumes from pod directory %q %+v", podDir, volumes)
+	klog.V(4).InfoS("Get volumes from pod directory", "podPath", podDir, "volumes", volumes)
```
```suggestion
	klog.V(4).InfoS("Get volumes from pod directory", "path", podDir, "volumes", volumes)
```

ehashman commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -171,11 +170,7 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoop() {
-			dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration),
-			dswp.getPodStatusRetryDuration)
-
+		klog.V(5).InfoS("Skipping findAndRemoveDeletedPods(). ", "nextRetryTime", dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration), "retryDuration", dswp.getPodStatusRetryDuration)
```
```suggestion
		klog.V(5).InfoS("Insufficient time has passed since last attempt, skipping findAndRemoveDeletedPods()", "nextRetryTime", dswp.timeOfLastGetPodStatus.Add(dswp.getPodStatusRetryDuration), "retryDuration", dswp.getPodStatusRetryDuration)
```

serathius replied:
Why resolve?

ehashman commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -182,18 +182,18 @@ func (rc *reconciler) unmountVolumes() {
 		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
 			// Volume is mounted, unmount it
-			klog.V(5).Infof(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
+			klog.V(5).InfoS(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
```
In another comment, @serathius asked for these to be refactored so we no longer need to call this GenerateMsgDetailed here. I agree but let's do that in a follow-up PR.

ehashman commented:
One other comment - for places where we don't have the pod object for KObj, we could use `klog.KRef(podNamespace, podName)` instead, as we should have both of those around...

ehashman commented:
Please ensure you squash your commits.

For one of the changes I suggested (typo fix of variable name), it required another change earlier in the code which is probably why the build is failing.

By the way, it might be a bit easier to split this PR into 2 or 3 parts, a +100 diff of all log lines is really challenging to review. I'd like to cap at 40 max per PR...

yangjunmyfm192085 commented:
/remove-area cloudprovider
/remove-area code-generation
/remove-area conformance
/remove-area dependency
/remove-area kubeadm
/remove-area kubectl
/remove-area provider/gcp
/remove-area release-eng
/remove-area test

yangjunmyfm192085 commented:
/remove-kind api-change

yangjunmyfm192085 commented:
/remove-sig  api-machinery
/remove-sig apps
/remove-sig architecture
/remove-sig  auth
/remove-sig cli
/remove-sig cloud-provider
/remove-sig cluster-lifecycle
/remove-sig instrumentation
/remove-sig network
/remove-sig release
/remove-sig storage
/remove-sig testing

yangjunmyfm192085 commented:
/remove-area  apiserver

yangjunmyfm192085 commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -278,19 +271,19 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 		exists, _, _ := dswp.actualStateOfWorld.PodExistsInVolume(volumeToMount.PodName, volumeToMount.VolumeName)
+		volumeToMountSpecName := "nil"
+		if volumeToMount.VolumeSpec != nil {
+			volumeToMountSpecName = volumeToMount.VolumeSpec.Name()
```
I have fixed it,thanks.

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-gce-storage-slow

yangjunmyfm192085 commented:
@ehashman I have fixed all and the pr is ready for review .

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -85,27 +85,27 @@ func (kl *Kubelet) runOnce(pods []*v1.Pod, retryDelay time.Duration) (results []
 		return results, fmt.Errorf("error running pods: %v", failedPods)
 	}
-	klog.Infof("%d pods started", len(pods))
+	klog.InfoS("Pods started", "podNum", len(pods))
```
```suggestion
	klog.InfoS("Pods started", "numPods", len(pods))
```
We should be consistent in naming arguments, above you use "numPods"

serathius commented on pkg/kubelet/runonce.go:
```diff
@@ -142,7 +142,7 @@ func (kl *Kubelet) runPod(pod *v1.Pod, retryDelay time.Duration) error {
 		}
 		// TODO(proppy): health checking would be better than waiting + checking the state at the next iteration.
-		klog.Infof("pod %q containers synced, waiting for %v", format.Pod(pod), delay)
+		klog.InfoS("Pod's containers synced, waiting", "pod", klog.KObj(pod), "waitingTime", delay)
```
I think we should use `duration` as this is more inline with name of the type.

serathius replied:
```suggestion
		klog.InfoS("Pod's containers synced, waiting", "pod", klog.KObj(pod), "duration", delay)
```

ehashman replied:
Please update `waitingTime` to `duration`

serathius commented on pkg/kubelet/volumemanager/BUILD:
```diff
@@ -1,93 +0,0 @@
-package(default_visibility = ["//visibility:public"])
```
All BUILD files should be already removed, please rebase to master branch.

yangjunmyfm192085 replied:
Thanks for reminding, I have fixed it. Thanks.

serathius commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -461,9 +461,7 @@ func (asw *actualStateOfWorld) addVolume(
-		klog.V(2).Infof("Volume %q is already added to attachedVolume list, update device path %q",
-			volumeName,
-			devicePath)
+		klog.V(2).InfoS("The volume is already added to attachedVolume list, update device path", "volumeName", volumeName, "devicePath", devicePath)
```
```suggestion
		klog.V(2).InfoS("The volume is already added to attachedVolume list, update device path", "volumeName", volumeName, "path", devicePath)
```

serathius commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -548,12 +544,7 @@ func (asw *actualStateOfWorld) MarkRemountRequired(
-					podObj.podUID,
-					volumeObj.volumeName,
-					podObj.volumeSpec.Name())
+				klog.ErrorS(nil, "MarkRemountRequired failed to FindPluginBySpec for volume", "podName", podObj.podName, "podUID", podObj.podUID, "volumeName", volumeName, "volumeSpecName", podObj.volumeSpec.Name())
```
```suggestion
				klog.ErrorS(nil, "MarkRemountRequired failed to FindPluginBySpec for volume", "pod", klog.KObj(podObj), "podUID", podObj.podUID, "volumeName", volumeName, "volumeSpecName", podObj.volumeSpec.Name())
```
Please apply fix or leave some comment. Resolving issues without doing so will result in re-posting.

yangjunmyfm192085 replied:
I tried to fix it,but failed. podObj is not work in klog.KObj(podObj).

serathius commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -530,9 +528,7 @@ func (asw *actualStateOfWorld) MarkVolumeAsResized(
-
-	klog.V(5).Infof("Volume %s(OuterVolumeSpecName %s) of pod %s has been resized",
-		volumeName, podObj.outerVolumeSpecName, podName)
+	klog.V(5).InfoS("Pod volume has been resized", "podName", podName, "volumeName", volumeName, "outerVolumeSpecName", podObj.outerVolumeSpecName)
```
I'm worried that the pod name passed here `podName` is not the type we usually want to put under this key. Please look at the type `volumetypes.UniquePodName`. Please either:
* [Preferred] Try to extract normal pod name of k8s object and use `podName` key.
* Use different key like `uniquePodName` to differentiate.

yangjunmyfm192085 replied:
Yes,you are right.
I have fixed all of them in this file according to "Use different key like uniquePodName to differentiate.".

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -525,11 +504,7 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
-			pod.Namespace,
-			pvcSource.ClaimName)
-
+		klog.V(5).InfoS("Found PVC", "pvc", klog.KRef(pod.Namespace, pvcSource.ClaimName))
```
```suggestion
		klog.V(5).InfoS("Found PVC", "PVC", klog.KRef(pod.Namespace, pvcSource.ClaimName))
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -548,14 +523,7 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
-			pvcUID,
-			pvName)
-
+		klog.V(5).InfoS("Found bound PV for PVC", "pvc", klog.KRef(pod.Namespace, pvcSource.ClaimName), "pvcUID", pvcUID, "pvName", pvName)
```
```suggestion
		klog.V(5).InfoS("Found bound PV for PVC", "PVC", klog.KRef(pod.Namespace, pvcSource.ClaimName), "PVCUID", pvcUID, "pvName", pvName)
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -566,15 +534,7 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
-			pvcSource.ClaimName,
-			pvcUID)
-
+		klog.V(5).InfoS("Extracted volumeSpec from bound PV and PVC", "pvc", klog.KRef(pod.Namespace, pvcSource.ClaimName), "pvcUID", pvcUID, "pvName", pvName, "volumeSpecName", volumeSpec.Name())
```
```suggestion
		klog.V(5).InfoS("Extracted volumeSpec from bound PV and PVC", "PVC", klog.KRef(pod.Namespace, pvcSource.ClaimName), "PVCUID", pvcUID, "pvName", pvName, "volumeSpecName", volumeSpec.Name())
```

serathius commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -63,11 +63,11 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 	volumePaths, err := kl.getMountedVolumePathListFromDisk(podUID)
 	if err != nil {
-		klog.Errorf("pod %q found, but error %v occurred during checking mounted volumes from disk", podUID, err)
+		klog.ErrorS(err, "The pod found, but error occurred during checking mounted volumes from disk", "podUID", podUID, "error", err)
```
```suggestion
		klog.ErrorS(err, "The pod found, but error occurred during checking mounted volumes from disk", "podUID", podUID)
```

serathius commented on pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:
```diff
@@ -558,14 +533,7 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
-			pvcUID,
-			pvName)
-
+		klog.V(5).InfoS("Found bound PV for PVC", "PVC", klog.KRef(pod.Namespace, pvcSource.ClaimName), "PVCUID", pvcUID, "pvName", pvName)
```
```suggestion
		klog.V(5).InfoS("Found bound PV for PVC", "PVC", klog.KRef(pod.Namespace, pvcSource.ClaimName), "PVCUID", pvcUID, "PVName", pvName)
```

serathius commented:
/lgtm

mrunalp commented on pkg/kubelet/container/runtime.go:
```diff
@@ -204,7 +204,7 @@ func BuildContainerID(typ, ID string) ContainerID {
 	var id ContainerID
 	if err := id.ParseString(containerID); err != nil {
-		klog.Error(err)
+		klog.ErrorS(err, "Parse containerID failed")
```
nit: `Parsing container ID failed`

mrunalp commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -63,11 +63,11 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 	volumePaths, err := kl.getMountedVolumePathListFromDisk(podUID)
 	if err != nil {
-		klog.Errorf("pod %q found, but error %v occurred during checking mounted volumes from disk", podUID, err)
+		klog.ErrorS(err, "The pod found, but error occurred during checking mounted volumes from disk", "podUID", podUID)
```
nit: Drop `The`

mrunalp commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -63,11 +63,11 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 	}
 	if len(volumePaths) > 0 {
-		klog.V(4).Infof("pod %q found, but volumes are still mounted on disk %v", podUID, volumePaths)
+		klog.V(4).InfoS("The pod found, but volumes are still mounted on disk", "podUID", podUID, "volumePaths", volumePaths)
```
Same here

mrunalp commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -86,7 +86,7 @@ func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *v1.Pod, o
 		return nil, fmt.Errorf("failed to instantiate mounter for volume: %s using plugin: %s with a root cause: %v", spec.Name(), plugin.GetPluginName(), err)
 	}
-	klog.V(10).Infof("Using volume plugin %q to mount %s", plugin.GetPluginName(), spec.Name())
+	klog.V(10).InfoS("Using the volume plugin for mount", "volumePluginName", plugin.GetPluginName(), "volumeName", spec.Name())
```
Drop `the`

mrunalp commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -118,7 +118,7 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 		// kl.getPodVolumePathListFromDisk(). Can this be cleaned up?
 		if podVolumesExist := kl.podVolumesExist(uid); podVolumesExist {
-			klog.V(3).Infof("Orphaned pod %q found, but volumes are not cleaned up", uid)
+			klog.V(3).InfoS("The orphaned pod found, but volumes are not cleaned up", "podUID", uid)
```
Drop `The`

mrunalp commented on pkg/kubelet/kubelet_volumes.go:
```diff
@@ -167,18 +167,18 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
+		klog.V(3).InfoS("The orphaned pod found, removing", "podUID", uid)
 		if err := removeall.RemoveAllOneFilesystem(kl.mounter, kl.getPodDir(uid)); err != nil {
-			klog.Errorf("Failed to remove orphaned pod %q dir; err: %v", uid, err)
+			klog.ErrorS(err, "Failed to remove the orphaned pod dir", "podUID", uid)
```
Drop `the`

yangjunmyfm192085 commented:
I have fixed all, thanks.@mrunalp

ehashman reviewed (commented):
/lgtm

Should be good for approval again

ehashman commented:
/hold cancel

ehashman commented:
/milestone v1.21

mrunalp commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -652,23 +649,23 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 			continue
 		}
-		klog.V(4).Infof("Volume: %s (pod UID %s) is marked as mounted and added into the actual state", volume.volumeName, volume.podName)
+		klog.V(4).InfoS("The volume is marked as mounted and added into the actual state", "podName", volume.podName, "volumeName", volume.volumeName)
```
Drop `The`

mrunalp commented on pkg/kubelet/volumemanager/reconciler/reconciler.go:
```diff
@@ -652,23 +649,23 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 				continue
 			}
-			klog.V(4).Infof("Volume: %s (pod UID %s) is marked device as mounted and added into the actual state", volume.volumeName, volume.podName)
+			klog.V(4).InfoS("The volume is marked device as mounted and added into the actual state", "podName", volume.podName, "volumeName", volume.volumeName)
```
Drop `The`

yangjunmyfm192085 commented:
I have fixed all @mrunalp.

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-kind

yangjunmyfm192085 commented:
/test pull-kubernetes-conformance-image-test

yangjunmyfm192085 commented:
/test pull-kubernetes-e2e-gce-csi-serial

yangjunmyfm192085 commented:
/test pull-kubernetes-conformance-image-test

ehashman commented on pkg/kubelet/volumemanager/cache/actual_state_of_world.go:
```diff
@@ -461,9 +461,7 @@ func (asw *actualStateOfWorld) addVolume(
-		klog.V(2).Infof("Volume %q is already added to attachedVolume list, update device path %q",
-			volumeName,
-			devicePath)
+		klog.V(2).InfoS("The volume is already added to attachedVolume list, update device path", "volumeName", volumeName, "path", devicePath)
```
Missed this `The`

```suggestion
		klog.V(2).InfoS("Volume is already added to attachedVolume list, update device path", "volumeName", volumeName, "path", devicePath)
```

yangjunmyfm192085 commented:
I have fixed them@ehashman

ehashman reviewed (commented):
/lgtm

pacoxu commented:
/retest
Need approval
